### PR TITLE
Better integration with cmake

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Automatically re-generate `.pb.h` and `.pb.cc` if `.proto` file changed, don't need to run `scripts/generate_cpp.sh` manually.

### CODE REVIEW

**Comment by windtalker** [comment]
LGTM

**Review by leiysky — APPROVED** [review]
LGTM

File: .gitignore [hunk 1]
```diff
@@ -16,6 +16,7 @@ _tools
 cpp/
 !cpp/CMakeLists.txt
 proto-cpp/
```
Comment (tisonkun) [inline .gitignore:18]:
```suggestion
```

Comment (JaySon-Huang) [reply .gitignore:18]: We still provide the script `scripts/generate-cpp.sh` for those project don't use CMake. I think we can keep this.

Comment (tisonkun) [reply .gitignore:18]: Well.

**Review by tisonkun — APPROVED** [review]
LGTM. Though it would be better if we wrapper actions into a function, as the official `FindProtobuf.cmake` does.

**Pull request merged**